src/openfermion/utils/_channel_state_test.py

```diff
@@ -0,0 +1,148 @@
+                                [0., 0., 0.5, -0.5],
+                                [0., 0., -0.5, 1.]])
+        self.assertAlmostEquals(norm(correct_matrix -
+                                     test_density_matrix), 0.0)
+
+        # Check for correct action on cat state
+        # With probability = 0
+        test_density_matrix = (
+            dephasing_channel(self.cat_matrix, 0, 1))
+        self.assertAlmostEquals(norm(self.cat_matrix -
+                                     test_density_matrix), 0.0)
+        # With probability = 1
+
+        correct_matrix = array([[0.50,  0.25,  0.00,  0.00],
+                                [0.25,  0.25, 0.00, -0.25],
+                                [0.00,  0.00,  0.00,  0.00],
+                                [0.00, -0.25, 0.00, 0.50]])
+        test_density_matrix = (
+            dephasing_channel(self.cat_matrix, 1, 1))
+        self.assertAlmostEquals(norm(correct_matrix -
+                                     test_density_matrix), 0.0)
+
+    def test_depolarizing(self):
+        """Test depolarizing on a simple qubit state"""
+
+        # With probability = 0
+        test_density_matrix = (
+            depolarizing_channel(self.cat_matrix, 0, 1))
+        self.assertAlmostEquals(norm(self.cat_matrix -
+                                     test_density_matrix), 0.0)
+
+        test_density_matrix = (
+            depolarizing_channel(self.cat_matrix, 0, 1,
+                                 transpose=True))
+        self.assertAlmostEquals(norm(self.cat_matrix -
+                                     test_density_matrix), 0.0)
+
+        # With probability 1 on both qubits
+        correct_density_matrix = array([[0.0555555, 0.,  0.0,  0.27777778],
+                                        [0.00, 0.00, -0.22222222, 0.000],
+                                        [0.00, -0.22222222, 0., 0.],
+                                        [0.27777778, 0.0, 0.0, 0.05555556]])
+
+        test_density_matrix = (
+            depolarizing_channel(self.cat_matrix, 1, 0))
+        test_density_matrix = (
+            depolarizing_channel(test_density_matrix, 1, 1))
+
+        self.assertAlmostEquals(norm(correct_density_matrix -
+                                     test_density_matrix), 0.0, places=6)
+
+        # Depolarizing channel should be self-adjoint
+        test_density_matrix = (
+            depolarizing_channel(self.cat_matrix, 1, 0,
+                                 transpose=True))
+        test_density_matrix = (
+            depolarizing_channel(test_density_matrix, 1, 1,
+                                 transpose=True))
+
+        self.assertAlmostEquals(norm(correct_density_matrix -
+                                     test_density_matrix), 0.0, places=6)
+
+    def test_verification(self):
+        """Verify basic sanity checking on inputs"""
+        with self.assertRaises(ValueError):
+            _ = amplitude_damping_channel(self.density_matrix, 2, 1)
```

Comment:
You likely don't need to do assignment for this, `amplitude_damping_channel(...)` would be fine - but it doesn't really matter